lib/vdb/vdb.cpp

```diff
@@ -0,0 +1,129 @@
+#include "pms-utils/vdb/vdb.hpp"
+
+#include "pms-utils/atom/atom.hpp"
+#include "pms-utils/depend/depend.hpp"
+#include "pms-utils/depend/depend_parser.hpp"
+#include "pms-utils/ebuild/ebuild.hpp"
+#include "pms-utils/misc/try_parse.hpp"
+#include "pms-utils/vdb/vdb_parsers.hpp"
+#include "util/readfile.hpp"
+
+#include <boost/parser/parser.hpp>
+#include <cstring>
+#include <expected>
+#include <filesystem>
+#include <format>
+#include <stdexcept>
+#include <string_view>
+
+using pms_utils::misc::try_parse;
+using pms_utils::misc::_internal::try_readfile;
+using pms_utils::vdb::_internal::Error;
+using pms_utils::vdb::_internal::IoError;
+using pms_utils::vdb::_internal::ParseError;
+
+namespace [[gnu::visibility("default")]] pms_utils {
+namespace vdb {
+
+namespace {
+
+template <typename T, typename E> T unwrap(std::expected<T, E> result, std::string_view message) {
+    if (!result) {
+        throw std::runtime_error(std::format("{}: {}", message, result.error()));
+    }
+
+    return result.value();
+}
+
+} // namespace
+
+Entry::Entry(const std::filesystem::path &path) {
+    {
+        for (const auto kind :
+             {std::string_view("DEPEND"), std::string_view("BDEPEND"), std::string_view("RDEPEND")}) {
```

Comment:
may want to consider https://en.cppreference.com/w/cpp/string/basic_string_view/operator%22%22sv ?